shelley/chain-and-ledger/formal-spec/chain.tex

```diff
@@ -713,191 +700,200 @@ \subsection{Block Header Transition}
   in the header (see Figure~\ref{fig:rules:bbody}).
 \end{itemize}
 
-The state update finally sets \var{slot} as new block header slot and hash of
-\var{bh} as the new block header hash in the block header state.
+If the size checks pass, then three transitions are done:
+
+\begin{itemize}
+  \item The $\mathsf{NEWEPOCH}$ transition performs any state change needed if it is the first
+    block of a new epoch.
+  \item The $\mathsf{UPDN}$ updates the evolving and candidate nonce.
+  \item The $\mathsf{RUPD}$ creates the reward update if it is late enough in the epoch.
+    \textbf{Note} that for every block header, either $\mathsf{NEWEPOCH}$ or $\mathsf{RUPD}$
+    will be the identity transition, and so, for instance, it does not matter if $\mathsf{RUPD}$
+    uses $\var{nes}$ or $\var{nes}'$ to obtain the needed state.
+\end{itemize}
 
 \begin{figure}[ht]
   \begin{equation}\label{eq:bheader}
     \inference[BHead]
     {
-      (\var{bhb},~\sigma) \leteq \var{bh}
-      &
-      \var{slot} \leteq \bslot{bhb}
-      \\
-      (\var{vk_{hot}},~\wcard,~\wcard,\var{c_0},~\wcard) \leteq \bocert{bhb}
-      & \var{hk} \leteq \hashKey{vk_{hot}}
-      \\
-      t \leteq \kesPeriod{slot} - c_0
-      \\~\\
-      \var{s_\ell} < \var{slot} \leq \var{s_{now}}
+      \var{bhb} \leteq \bhbody{bh}
       &
-      \var{h} = \bprev{bhb}
+      \var{s} \leteq \bslot{bhb}
       &
-      \mathcal{V}^{\mathsf{KES}}_{vk_{hot}}{\serialised{bhb}}_{\sigma}^{t}
+      \eta \leteq \fun{bnonce}~\var{bhb}
       \\
       \bHeaderSize{bh} \leq \fun{maxHeaderSize}~\var{pp}
       &
       \hBbsize{bhb} \leq \fun{maxBlockSize}~\var{pp}
+      \\
+      {
+        \left(
+          {\begin{array}{c}
+              \eta_c \\
+              \var{s} \\
+              \var{gkeys} \\
+          \end{array}}
+        \right)
+        \vdash
+        \var{nes}
+        \trans{\hyperref[fig:rules:new-epoch]{newepoch}}{\epoch{s}}
+        \var{nes}'
+      }
+      \\
+      {
+        \eta \vdash
+        {\left(\begin{array}{c}
+              \eta_v \\
+              \eta_c \\
+        \end{array}\right)}
+        \trans{\hyperref[fig:rules:update-nonce]{updn}}{\var{s}}
```

Comment:
I wonder whether it would make more sense for `updn` to live inside `PRTCL`? Nothing else needs to know about the nonces, and I generally see them as part of Praos rather than the ledger - in particular, I would expect them to be implemented in the consensus layer.

Reply:
yes, that's much better, thanks!